File: designer/src/main/java/com/skm/apachepoi/designer/DesignerHook.java
package com.skm.apachepoi.designer;

import com.inductiveautomation.ignition.common.script.ScriptManager;
import com.inductiveautomation.ignition.common.script.hints.PropertiesFileDocProvider;
import com.inductiveautomation.ignition.designer.model.AbstractDesignerModuleHook;
import com.skm.apachepoi.client.ClientScriptModule;

public class DesignerHook extends AbstractDesignerModuleHook {

@Override
public void initializeScriptManager(ScriptManager manager) {
  super.initializeScriptManager(manager);

  manager.addScriptModule(
      "system.excel",
      new ClientScriptModule(),
      new PropertiesFileDocProvider());
}

}

File: designer/src/main/kotlin/dev/scottkey/sftp/designer/DesignerHook.kt


package dev.scottkey.sftp.designer

import com.inductiveautomation.ignition.common.script.ScriptManager
import com.inductiveautomation.ignition.common.script.hints.PropertiesFileDocProvider
import com.inductiveautomation.ignition.designer.model.AbstractDesignerModuleHook
import dev.scottkey.sftp.client.ClientScriptModule
import dev.scottkey.sftp.modulePath


@Suppress("unused")
class DesignerHook : AbstractDesignerModuleHook() {
    override fun initializeScriptManager(manager: ScriptManager) {
        super.initializeScriptManager(manager)
        manager.addScriptModule(
            modulePath,
            ClientScriptModule(),
            PropertiesFileDocProvider()
        )
    }
}